### PR TITLE
Migrate to Dart 3 and Flutter 3.10.0

### LINKED ISSUE #451: Migrate to Dart 3
Migration guide:
https://dart.dev/resources/dart-3-migration

Blog post:
https://medium.com/dartlang/announcing-dart-3-53f065a10635

### PR DESCRIPTION
**Pull request by marfavi**
Closes #451

### CODE REVIEW

**Comment by marfavi** [comment]
@TTA777 Can you generate some new goldens on this branch? 🙏🏽

**Comment by TTA777** [comment]
@marfavi Done

File: analysis_options.yaml [hunk 1]
```diff
@@ -26,25 +24,23 @@ analyzer:
...
   rules:
     - avoid-ignoring-return-values:
         exclude:
           - test/**
-
     - no-empty-block:
         exclude:
           - test/**
-
-    - list-all-equatable-fields
```
Comment (fremartini) [inline analysis_options.yaml:46]: is this not needed anymore?

**Review by fremartini — DISMISSED** [review]

File: analysis_options.yaml (hunk 1 above)
Comment (marfavi) [reply analysis_options.yaml:46]: My bad. I'll re-add it

**Review by fremartini — APPROVED** [review]

**Pull request merged**